packages/lesswrong/lib/utils/schemaUtils.ts

```diff
@@ -133,7 +133,18 @@ export function arrayOfForeignKeysField<CollectionName extends keyof Collections
   
   return {
     type: Array,
+
     defaultValue: [],
+    onCreate: ({newDocument, fieldName}: {
+      newDocument: DbObject,
+      fieldName: string,
```

Comment:
The types here aren't very pretty, leading to the cast below.

You could use CollectionName?

Reply:
Unfortunately, no, because there are two collections involved (this is setting up a foreign-key relation) and the `CollectionName` type is for the wrong collection. The `arrayOfForeignKeysField` function only takes one of the two collection names (the other is the collection this field is added two, so the one we need isn't available.